docs/testing-guide/opcodes.md

```diff
@@ -0,0 +1,363 @@
+            assert exists
+            return balance
+
+    contract = AccountParamsContract()
+    balance = contract.get_account_balance(account)
+    assert balance == algopy.UInt64(1000000)
+```
+
+### algopy.op.AppParamsGet
+
+```{testcode}
+from algopy_testing import algopy_testing_context
+
+class AppParamsContract(algopy.ARC4Contract):
+    @algopy.arc4.abimethod
+    def get_app_creator(self, app_id: algopy.Application) -> algopy.arc4.Address:
+        creator, exists = algopy.op.AppParamsGet.app_creator(app_id)
+        assert exists
+        return creator
+
+
+with algopy_testing_context() as ctx:
+    contract = AppParamsContract()
+    app = ctx.any.application()
+    creator = contract.get_app_creator(app)
+    assert creator == ctx.default_sender
+```
+
+### algopy.op.AssetParamsGet
+
+```{testcode}
+from algopy_testing import algopy_testing_context
+import algopy.op as op
+
+with algopy_testing_context() as ctx:
+    asset = ctx.any.asset(total=algopy.UInt64(1000000), decimals=algopy.UInt64(6))
+
+    class AssetParamsContract(algopy.ARC4Contract):
+        @algopy.arc4.abimethod
+        def get_asset_total(self, asset_id: algopy.UInt64) -> algopy.UInt64:
+            total, exists = op.AssetParamsGet.asset_total(asset_id)
+            assert exists
+            return total
+
+    contract = AssetParamsContract()
+    total = contract.get_asset_total(asset.id)
+    assert total == algopy.UInt64(1000000)
+```
+
+### algopy.op.Box
+
+```{testcode}
+from algopy_testing import algopy_testing_context
+import algopy.op as op
+
+with algopy_testing_context() as ctx:
+    class BoxStorageContract(algopy.ARC4Contract):
+        @algopy.arc4.abimethod
+        def store_and_retrieve(self, key: algopy.Bytes, value: algopy.Bytes) -> algopy.Bytes:
+            op.Box.put(key, value)
+            retrieved_value, exists = op.Box.get(key)
+            assert exists
+            return retrieved_value
+
+    contract = BoxStorageContract()
+    key, value = algopy.Bytes(b"test_key"), algopy.Bytes(b"test_value")
+    result = contract.store_and_retrieve(key, value)
+    assert result == value
+    stored_value = ctx.ledger.get_box(contract, key)
+    assert stored_value == value.value
+```
+
+## Mockable Opcodes
+
+These opcodes are mockable in `algorand-python-testing`, allowing for controlled testing of complex operations:
+
+### algopy.compile_contract
+
+```{testcode}
+import unittest
+from unittest.mock import patch, MagicMock
+import algopy
+from algopy_testing.primitives import Bytes
+
+mocked_response = MagicMock()
+mocked_response.approval_program = (Bytes(b'mock_approval'), Bytes(b'mock_approval'))
+mocked_response.clear_state_program = (Bytes(b'mock_clear'), Bytes(b'mock_clear'))
+
+class MockContract(algopy.Contract):
+    pass
+
+with patch('algopy.compile_contract', return_value=mocked_response) as mock_compile_contract:
+    compiled = algopy.compile_contract(MockContract)
```

Comment:
Would be better to show mocking this function while executing a contract method that uses it, rather than calling the function directly in a test